src/misc/graphicality.c

```diff
@@ -627,17 +603,35 @@ static igraph_error_t igraph_i_is_graphical_directed_simple(const igraph_vector_
         return IGRAPH_SUCCESS;
     }
 
-    /* Create an index vector that sorts the vertices by decreasing in-degree */
-    IGRAPH_CHECK(igraph_vector_int_init_range(&index_array, 0, vcount));
-    IGRAPH_FINALLY(igraph_vector_int_destroy, &index_array);
+    
+    IGRAPH_CHECK(igraph_vector_int_list_init(&buckets, vcount));
+    IGRAPH_FINALLY(igraph_vector_int_list_destroy, &buckets);
 
-    /* Set up the auxiliary struct for sorting */
-    sort_data.first  = in_degrees;
-    sort_data.second = out_degrees;
+    for (i = 0; i < vcount; i++) {
+        if (VECTOR(*in_degrees)[i] >= vcount || VECTOR(*out_degrees)[i] >= vcount) {
+            *res = false;
+            igraph_vector_int_list_destroy(&buckets);
+            IGRAPH_FINALLY_CLEAN(1);
+            return IGRAPH_SUCCESS;
+        }
+        current_bucket = igraph_vector_int_list_get_ptr(&buckets, VECTOR(*in_degrees)[i]);
+        igraph_vector_int_push_back(current_bucket, VECTOR(*out_degrees)[i]);
```

Comment:
We are probably missing an `IGRAPH_CHECK` here (unless you can ensure that the operation never needs to reallocate the underlying memory, in which case this should be documented).

Reply:
That is correct, will fix